### DISCUSSION TITLE
'find' command suddenly not working in windows hosted agents

### DISCUSSION THREAD

**Discussion by dumbledad** [community/community#25042] [Actions]
<p>In the YAML file defining our ADO build pipeline we have the following section</p>
<pre><code>- bash: |
    pip install pylint
    find preprocessing/ subtyping/ visualisations/ -name '*.py' | grep -v hmm | grep -v /test_ | sort | xargs pylint
  displayName: 'Run linter'
</code></pre>
<p>At some point in the last week this stopped working. It still works on ubuntu-latest hosted agent, but on Windows (either windows-latest or vs2017-win2016) it now fails with the message</p>
<p>&gt; FIND: Parameter format not correct</p>
<p>We have not changed anything, but our buld process now fails.</p>
<ol>
<li>What’s changed to cause this failure?</li>
<li>Why does it only fail on Windows?</li>
<li>How might I diagnose the problem?</li>
<li>Is there anything wrong with our call to bash’s ‘find’ command?</li>
</ol>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/dumbledad">@dumbledad</a> ,</p>
<p>This could be caused by the github hosted runner update. in bash shell, cannot locate correct ‘find’, if you type ‘which find’ to check, it will hint ‘/c/windows/system32/find’. However, the correct path should be ‘/usr/bin/find’.</p>
<p>You can use ‘/usr/bin/find’ instead for the command. Please check similar <a href="https://github.community/t/find-not-working-with-latest-runner/17413">ticket</a> for more details.</p>
<p>Thanks.</p>
